src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -173,10 +175,10 @@ void RegisterContainerAdapter(IContainer containerToAdapt)
                 await installer.Install(username).ConfigureAwait(false);
             }
         }
-
+        
         Task CreateQueuesIfNecessary(TransportReceiveInfrastructure receiveInfrastructure, string username)
         {
-            if (!settings.CreateQueues())
+            if (!ShouldRunInstallers || !settings.CreateQueues())
```

Comment:
This logic is a bit mind-bending. At this point I'd prefer to see `if(runInstallers and createQueues) { logic }'